### PR TITLE
Fix changelog section hyperlinks

### LINKED ISSUE #5637: Changelog links don't reference specific release version
Duplicate of https://github.com/dask/dask/issues/8473

Summary:
Anchor links in the changelog don't actually reference a specific released version.
If I go to the changelog and click on the `2021.12.0` link, I'm redirected to https://docs.dask.org/en/stable/changelog.html#id1 when, naively, I would have expected this link to look like https://docs.dask.org/en/stable/changelog.html#2021.12.0

### PR DESCRIPTION
**Pull request by aneeshnema**
- [x] Closes #5637
- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

This MR:
- ~changes version titles from `2021.12.0` to `v2021.12.0`~
- generates links like `#v2021.12.0` and `#v2-30-0-2020-10-06` instead of #id1, #id2

Duplicate of https://github.com/dask/dask/pull/8534

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by aneeshnema** [comment]
// @jrbourbeau

File: docs/source/changelog.rst [hunk 1]
```diff
@@ -2293,12 +2495,16 @@ This is a small bugfix release due to a config change upstream.
 -  Add Client.current() (:pr:`1450`)
 -  Clean handling of restart timeouts (:pr:`1442`)
 
+.. _v1.19.1 - September 25th, 2017:
```
Comment (jrbourbeau) [inline docs/source/changelog.rst:2498]: Thanks @aneeshnema! This results in links that look like `...changelog.html#september-25th-2017`. Could you update this link and section title, and the occurrences below, to all use the YYYY-MM-DD format similar to what's used above?

Comment (aneeshnema) [reply docs/source/changelog.rst]: done

**Review by jrbourbeau — APPROVED** [review]
Thanks @aneeshnema! This looks great

**Pull request merged**